src/category_theory/groupoid/subgroupoid.lean

```diff
@@ -537,6 +539,92 @@ lemma is_normal_map (hφ : function.injective φ.obj) (hφ' : im φ hφ = ⊤) (
 
 end hom
 
+section thin
+
+/-- A subgroupoid `is_thin` if it has at most one arrow between any two vertices. -/
+abbreviation is_thin := is_thin S.objs
+
+lemma is_thin_iff : S.is_thin ↔ ∀ (c d : S.objs), subsingleton (S.arrows c d) :=
+⟨λ h c d, h c d, λ h c d, h c d⟩
```

Comment:
```suggestion
lemma is_thin_iff : S.is_thin ↔ ∀ (c d : S.objs), subsingleton (S.arrows c d) :=
rfl
```

Reply:
Also, should `groupoid.is_thin_iff` be ported here as well?

Reply:
For this and the other `rfl` suggestion, I get 
```
type mismatch, term
  rfl
has type
  ?m_2 = ?m_2
but is expected to have type
```
so I don't think `rfl` works?

As for porting `groupoid.is_thin_iff`, I put it in `basic.lean` because it doesn't have to do with *sub*groupoids. Does that not make sense?

Reply:
>  I don't think `rfl` works?

Sorry. It should be `iff.rfl` instead.
> As for porting groupoid.is_thin_iff, I put it in basic.lean because it doesn't have to do with subgroupoids. Does that not make sense?

I meant the following lemma:
```
lemma is_thin_iff : S.is_thin ↔ ∀ (c ∈ S.objs), subsingleton (S.arrows c c) := sorry
```

Reply:
Alright, I'll do that tomorrow morning. Thanks for the review!